test/unit/controllers/webhook.test.js

```diff
@@ -0,0 +1,751 @@
+let mockCreateHmacFn = jest.fn()
+
+const helpers = require('./../../helpers')
+const Review = require('../../../lib/models/Review')
+const sampleData = require('./../../helpers/sampleData')
+
+let req
+let res
+let mockReview
+
+let mockGetReviewFn = jest.fn()
+let mockDeleteBranchFn = jest.fn()
+let mockCreateFn = jest.fn()
+let mockSetConfigPathFn = jest.fn()
+let mockGetSiteConfigFn = jest.fn()
+let mockProcessMergeFn = jest.fn()
+
+jest.mock('../../../lib/GitServiceFactory', () => {
+  return {
+    create: mockCreateFn
+  }
+})
+
+const mockHmacDigest = 'mock Hmac digest'
+/*
+ * Mock the createHmac function within the native crypto module, but leave every other function.
+ * This allows us to test logic that invokes GitHub's webhook request authentication, which
+ * involves signing the webhook payload using the agreed-upon secret token.
+ */
+jest.mock('crypto', () => {
+  const cryptoOrig = require.requireActual('crypto')
+  return {
+    ...cryptoOrig,
+    createHmac: mockCreateHmacFn
+  }
+})
+
+jest.mock('../../../lib/Staticman', () => {
+  return jest.fn().mockImplementation(() => {
+    return {
+      setConfigPath: mockSetConfigPathFn,
+      getSiteConfig: mockGetSiteConfigFn,
+      processMerge: mockProcessMergeFn
+    }
+  })
+})
+
+// Instantiate the module being tested AFTER mocking dependendent modules above.
+const webhook = require('./../../../controllers/webhook')
+
+beforeEach(() => {
+  req = helpers.getMockRequest()
+  res = helpers.getMockResponse()
+
+  mockCreateFn.mockImplementation((service, options) => {
+    return {
+      getReview: mockGetReviewFn,
+      deleteBranch: mockDeleteBranchFn
+    }
+  })
+
+  mockCreateHmacFn.mockImplementation((algo, data) => {
+    return {
+      update: data => {
+        return {
+          digest: encoding => {
+            return mockHmacDigest
+          }
+        }
+      }
+    }
+  })
+})
+
+afterEach(() => {
+  mockGetReviewFn.mockClear()
+  mockDeleteBranchFn.mockClear()
+  mockCreateFn.mockClear()
+  mockSetConfigPathFn.mockClear()
+  mockGetSiteConfigFn.mockClear()
+  mockProcessMergeFn.mockClear()
+  mockCreateHmacFn.mockClear()
+})
```

Comment:
```suggestion
afterEach(() => {
  jest.clearAllMocks()
```

Would this work instead?